components/src/core/header/headerContent.tsx

```diff
@@ -215,9 +215,15 @@ export const HeaderContent: React.FC<HeaderContentProps> = (props) => {
     const { headerHeight } = useHeaderHeight();
     const { searching, searchConfig } = useSearch();
     const defaultStyles = headerContentStyles;
+    const [collapsed, setCollapsed] = useState(true);
 
     let content: JSX.Element[] = [];
 
+    useEffect(() => {
+        const listener = headerHeight.addListener(({ value }) => setCollapsed(value === REGULAR_HEIGHT));
+        headerHeight.removeListener(listener);
```

Comment:
my PR eliminates the need for this function entirely, but wanted to point out that you need to return a function in this hook in order to clean up the listener. What you have now destroys the listener immediately (collapsed always stays false).
```suggestion
        return () => headerHeight.removeListener(listener);
```